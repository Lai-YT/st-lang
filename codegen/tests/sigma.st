% Sigma.st
%
% Compute sum = 1 + 2 + ... + n
%

% constant and variable declarations
const n: int := 10
var sum: int
var index: int

% with for statement
sum := 0
for index: 1 .. n
  sum := sum + index
end for
put "The result is ", sum

skip

% with loop statement
sum := 0
index := 1
loop
  sum := sum + index
  index := index + 1
  exit when index > n
end loop
put "The result is ", sum
