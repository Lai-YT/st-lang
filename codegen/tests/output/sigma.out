The result is 55

The result is 55
